synapse/metrics/background_process_metrics.py

```diff
@@ -216,6 +247,7 @@ async def run():
                     "Background process '%s' threw an exception",
                     desc,
                 )
+                raise
```

Comment:
We're missing a `return` or a `raise` in this branch.
raising here makes looping calls stop, so we should return `None` instead.